pkg/kubectl/cmd/delete.go

```diff
@@ -69,10 +69,14 @@ Examples:
 				NamespaceParam(cmdNamespace).DefaultNamespace().
 				FilenameParam(flags.Filenames...).
 				SelectorParam(cmdutil.GetFlagString(cmd, "selector")).
-				ResourceTypeOrNameArgs(args...).
+				SelectedResourceTypeOrNameArgs(args...).
 				Flatten().
 				Do()
-			checkErr(r.Err())
+
+			if r.Err() != nil {
```

Comment:
Ah - this is a more general problem with our client tools.  Since we have this problem everywhere in kubectl (we can't test failures) I'd probably prefer to use `checkErr` for here and do the test in hack/test-cmd.sh (which can test failure).  We do need to correct that problem in kubectl eventually, but I'm a bit worried about having two different ways of checking errors start creeping into the code.